MapleServer2/PacketHandlers/Game/ClubHandler.cs

```diff
@@ -66,18 +73,54 @@ private void HandleJoin(GameSession session, PacketReader packet)
             }
 
             long clubId = packet.ReadLong();
-            int response = packet.ReadInt(); // 0 = accept, 76 (0x4C) = reject??? lol
 
-            string clubName = ""; // TODO get club name from club
+            Club club = GameServer.ClubManager.GetClubById(clubId);
+            if (club == null)
+            {
+                return;
+            }
 
-            // TODO handle rejections
+            if (club.Leader.CharacterId == session.Player.CharacterId)
+            {
+                foreach (Player member in party.Members)
+                {
+                    party.BroadcastPacketParty(ClubPacket.UpdatePlayerClubList(member, club.Id));
+                }
+            }
+            else
+            {
+                // you are not leader but have responded to the invitation
```

Comment:
You can remove this comment, especially since the "you are the leader" comment above is not here